src/System.Resources.Writer/src/System/Resources/ResourceWriter.cs

```diff
@@ -418,8 +419,11 @@ public void Generate()
                 bw.Write(startOfDataSection);
 
                 // Write name section.
-                nameSection.Seek(0, SeekOrigin.Begin);
-                nameSection.CopyTo(bw.BaseStream, (int)nameSection.Length);
+                if (nameSection.Length > 0)
+                {
+                    nameSection.Seek(0, SeekOrigin.Begin);
+                    nameSection.CopyTo(bw.BaseStream, (int)nameSection.Length);
+                }
```

Comment:
This is fine, but I'm curious why this was changed away from the original code as was in desktop that used Write instead of CopyTo?